subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ArtifactTransformOutputsInternal.java

```diff
@@ -24,15 +24,18 @@
 
 public interface ArtifactTransformOutputsInternal extends ArtifactTransformOutputs {
 
-    static void validateOutputLocation(File output, File primaryInput, String primaryInputPrefix, File outputDir, String outputDirPrefix) {
-        if (output.equals(primaryInput) || output.equals(outputDir)) {
-            return;
+    static OutputLocationType validateOutputLocation(File output, File primaryInput, String primaryInputPrefix, File outputDir, String outputDirPrefix) {
```

Comment:
Nit: method is not about validation anymore, it's just a side-effect of determining the type of output, how about `determineOutputLocationType()`?

Reply:
Renamed.